src/pages/umi/web3js-differences-and-adapters.md

```diff
@@ -0,0 +1,377 @@
+// Generate a new Web3Js Publickey
+const web3jsPublickey = new PublicKey("<1111111111111111111111111111111111111111>");
+```
+
+Next, let's look into how to use the adapters.
+
+### From Web3Js to Umi
+```ts
+import { PublicKey } from '@solana/web3.js';
+import { fromWeb3JsPublicKey } from '@metaplex-foundation/umi-web3js-adapters';
+
+// Generate a new Publickey
+const web3jsPublickey = new PublicKey("<1111111111111111111111111111111111111111>");
+
+// Convert it using the UmiWeb3jsAdapters Package
+const umiPublicKey = fromWeb3JsPublicKey(web3jsPublickey);
+```
+
+### From Umi to Web3Js
+```ts
+import { publicKey } from '@metaplex-foundation/umi';
+import { toWeb3JsPublicKey } from '@metaplex-foundation/umi-web3js-adapters';
+
+// Generate a new Publickey
+const umiPublicKey = publicKey("11111111111111111111111111111111");
+
+// Convert it using the UmiWeb3jsAdapters Package
+const web3jsPublickey = toWeb3JsPublicKey(umiPublicKey);
+```
+
+## Keypairs
+
+Generating keypairs is where the difference from Web3Js and Umi increase. For Web3Js we can just use the `Keypair.generate()` function, for Umi we need a Umi instance before.
+
+### Umi
+```ts
+import { createUmi } from '@metaplex-foundation/umi-bundle-defaults'
+import { generateSigner, createSignerFromKeypair } from '@metaplex-foundation/umi'
+
+// Generate a new Umi instance
+const umi = createUmi('https://api.devnet.solana.com')
+
+// Generate a new Umi keypair
+const umiKeypair = generateSigner(umi)
+
+// Or use an existing one
+const umiKeypair = umi.eddsa.createKeypairFromSecretKey(new Uint8Array(walletFile));
+```
+
+### Web3Js
+```ts
+import { Keypair } from '@solana/web3.js';
+
+// Generate a new Web3Js keypair
+const web3jsKeypair = Keypair.generate();
+
+// Or use an existing one
+const web3jsKeypair = Keypair.fromSecretKey(new Uint8Array(walletFile));
+```
+
+Next, let's look into how to use the adapters.
+
+### From Web3Js to Umi
+```ts
+import { Keypair } from '@solana/web3.js';
+import { fromWeb3JsKeypair } from '@metaplex-foundation/umi-web3js-adapters';
+
+// Generate a new keypair
+const web3jsKeypair = Keypair.generate();
+
+// Convert it using the UmiWeb3jsAdapters Package
+const umiKeypair = fromWeb3JsKeypair(web3jsKeypair);
+```
+
+### From Umi to Web3Js
+```ts
+import { createUmi } from '@metaplex-foundation/umi-bundle-defaults'
+import { generateSigner } from '@metaplex-foundation/umi'
+import { toWeb3JsKeypair } from '@metaplex-foundation/umi-web3js-adapters';
+
+// Generate a new Umi instance
+const umi = createUmi('https://api.devnet.solana.com')
+
+// Generate a new keypair
+const umiKeypair = generateSigner(umi)
+
+// Convert it using the UmiWeb3jsAdapters Package
+const web3jsKeypair = toWeb3JsKeypair(umiKeypair);
+```
+
+## Instructions
+
+When creating instructions, the main difference is that with Umi, you need to create a Umi instance first (like the `Keyapair`) and using the `getInstructions()` you receive an array of instructions rather than a single one.
```

Comment:
```suggestion
When creating instructions, the main difference is that with Umi, you need to create a Umi instance first (like the `Keypair`) and using the `getInstructions()` you receive an array of instructions rather than a single one.
```